### PR TITLE
Set up foundation for automated application processing

### PR DESCRIPTION
**Pull request by jodyheavener**
`dev/-/epics/529`

## Summary

We are working on improvements to the 1Password for Open Source program. This PR sets up a foundation for an eventual application review and approval process:

- Sets up a new Go package, `1password-open-source-processor`, that will eventually be expanded on in future PRs such that it can be used in GitHub workflows for reviewing and processing applications.
- Also sets up a Makefile for installing deps, building the script, tagging new versions
- Sets up a GitHub workflow that, when a `processor-*` tag is pushed:
  1. Parses the pushed tag version
  2. Builds the processor script
  3. Creates a new release with the script as an asset under the parsed version

The idea here is that we don't want to have to build the Go script every time an application is reviewed or approved, so we'll just build it one time and workflows will download and use the pre-built version via the latest release.

### CODE REVIEW

File: .github/workflows/build-processor.yml [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+permissions:
+  contents: write
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v3
```
Comment (MNThomson) [inline .github/workflows/build-processor.yml:17]: [Node.js 16 actions are deprecated](https://github.blog/changelog/2023-09-22-github-actions-transitioning-from-node-16-to-node-20/) (which `v3` uses), lets upgrade to `v4`
```suggestion
        uses: actions/checkout@v4
```

File: .github/workflows/build-processor.yml [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+
+      - name: Setup Go
+        uses: actions/setup-go@v2
+        with:
+          go-version: 1.19
```
Comment (MNThomson) [inline .github/workflows/build-processor.yml:22]: Any specific reason to be using `v2` of the action, looks like `v5` is the current version? Upgrading provides some additional features like retrieving the go version from the mod file & dependency caching (not sure how this action will deal with a non-existant `go.sum`, might be better to add this once the `go.sum` exists).
```suggestion
      - name: Setup Go
        uses: actions/setup-go@v5
        with:
          go-version-file: 'script/go.mod'
          cache-dependency-path: 'script/go.sum'
```

**Review by MNThomson — APPROVED** [review]
LGTM after those two action version upgrades!

File: .github/workflows/build-processor.yml (hunk 2 above)
Comment (MNThomson) [reply .github/workflows/build-processor.yml:22]: [Looks like the file not existing shouldn't be a problem](https://github.com/actions/setup-go?tab=readme-ov-file#caching-dependency-files-and-build-outputs):
> If some problem that prevents success caching happens then the action issues the warning in the log and continues the execution of the pipeline.

Comment (jodyheavener) [reply .github/workflows/build-processor.yml:22]: Indeed, looks like it works with a simple warning.

CleanShot 2024-03-13 at 07 37 02@2x

Good thinking!

Comment (jodyheavener) [reply .github/workflows/build-processor.yml:22]: Thanks for catching these outdated versions.

**Pull request merged**